### ISSUE TITLE
[Q] Is there a way to launch and log multiple runs and sync at the end?

### ISSUE THREAD

**Issue by sisaman** [wandb/client#1953]
I am launching multiple runs within a single script by setting reinit=True when initializing wandb and calling run.finish() at the end according to the documentation here:
https://docs.wandb.ai/library/init#how-do-i-launch-multiple-runs-from-one-script

However, each call to run.finish() takes around 3 secs on my machine, which is inefficient as my runs end pretty quickly (each run logs between 500-1000 scalar values in about 5 secs), and I am launching hundreds of them. So I was wondering if there's any other way to do it more efficiently?

I think I can also use offline mode and sync at the end, but I don't know how to do it in the script itself (not manually with the command line).

**Comment by vanpelt** [comment]
Hey @sisaman wandb doesn't support running many really short scripts well today.  Your best option is to use dryrun mode then just write a script that calls `subprocess.check_call(["wandb", "sync", "."])` once you've finished.

**Comment by sisaman** [comment]
Even in the dryrun mode, run.finish() still takes a few seconds. Is that normal?

**Comment by vanpelt** [comment]
Yeah, we need to flush data to disk and restore console output.  If you just want to log a bunch of runs with summary data and not media / history you could create the runs using the public API which is far less overhead, i.e.

```
api = wandb.Api()
run = api.create_run()
run.config.update({"learning_rate": 0.3})
run.summary.update({"acc": 1, "val_acc": 3})
```

**Comment by sisaman** [comment]
That works better, thank you!

**Comment by evfro** [comment]
```python
def api_run():
    api = wandb.Api()
    run = api.create_run()
    run.config.update({"learning_rate": 0.3})
    run.summary.update({"acc": 1, "val_acc": 3})

wandb.agent(sweep_id, function=api_run)
```

@vanpelt  strangely, if I run this snippet, runs remain in a "running" state forever and do not update runs table with the results.  Not sure what's the problem here...

**Comment by evfro** [comment]
@vanpelt  yeah, that's what I did in the first place.
It works, but keeps printing the `signal only works in main thread of the main interpreter` warning at each run.
![image](https://user-images.githubusercontent.com/5283394/116604490-df9f0180-a936-11eb-84bc-709d90ce2b0d.png)

Even though it's not an error, I still feel uncomfortable with it.  So I started exploring other options.

The reason for this setup is that I use my own grid search scheduler. Wandb is only used for logging the results.
